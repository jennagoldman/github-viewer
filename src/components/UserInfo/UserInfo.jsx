import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ name, followerCount, profileLink }) => (
  <div>
    <span>{name}</span>
    <span>{followerCount} followers</span>
    <a href={profileLink}>View profile</a>
  </div>
);

UserInfo.propTypes = {
  name: PropTypes.string.isRequired,
  followerCount: PropTypes.string.isRequired,
  profileLink: PropTypes.string.isRequired,
};

export default UserInfo;
