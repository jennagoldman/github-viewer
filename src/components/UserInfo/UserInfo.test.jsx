import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from './UserInfo.jsx';

describe('UserInfo', () => {
  it('matches a snapshot', () => {

    const repos = [
      { id: 1,
        name: 'rock-paper-scissors',
        html_url: 'https://github.com/jennagoldman/rock-paper-scissors' }
    ];

    const wrapper = shallow(<UserInfo name="Jenna Goldman" followerCount="2" profileLink="https://github.com/jennagoldman" repos={repos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
