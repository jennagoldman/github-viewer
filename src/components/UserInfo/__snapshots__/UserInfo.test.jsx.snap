// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo matches a snapshot 1`] = `
<div>
  <p>
    Jenna Goldman
  </p>
  <p>
    2
     followers
  </p>
  <a
    href="https://github.com/jennagoldman"
    rel="noopener noreferrer"
    target="_blank"
  >
    View 
    Jenna Goldman
    's GitHub Profile
  </a>
  <Repos
    repos={
      Array [
        Object {
          "html_url": "https://github.com/jennagoldman/rock-paper-scissors",
          "id": 1,
          "name": "rock-paper-scissors",
        },
      ]
    }
  />
</div>
`;
